### PR TITLE
Profile page including profile photos

### PR DESCRIPTION
**Pull request by Macludde**

### CODE REVIEW

File: src/routes/member/[studentId]/+page.server.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
+import prisma from "$lib/utils/prisma";
+import { error, fail } from "@sveltejs/kit";
+import type { PageServerLoad } from "./$types";
+import { ctxAccessGuard } from "$lib/utils/access";
+import apiNames from "$lib/utils/apiNames";
+import { Prisma } from "@prisma/client";
+import { _classProgrammes } from "./data";
+
+export const load: PageServerLoad = async ({ params }) => {
+  const [member, publishedArticles] = await Promise.all([
```
Comment (danieladugyan) [inline src/routes/member/[studentId]/+page.server.ts:10]: Apparently you're [supposed to avoid Promise.all](https://chrysanthos.xyz/article/dont-ever-use-promise-all/)

File: src/routes/member/[studentId]/+page.svelte [hunk 2]
```diff
@@ -0,0 +1,233 @@
+<script lang="ts">
```
Comment (danieladugyan) [inline src/routes/member/[studentId]/+page.svelte:1]: These pages do get a bit long. I think we should consider composing components, either by just splitting stuff up into components, or additionally using slots.

File: src/routes/member/[studentId]/edit-bio/+page.server.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    member,
+  };
+};
+
+export const actions = {
+  update: async ({ params, locals, request }) => {
+    const session = await locals.getSession();
+    await ctxAccessGuard(apiNames.MEMBER.UPDATE, session?.user, { studentId: params.studentId });
+    const studentId = params.studentId;
+    const formData = await request.formData();
```
Comment (danieladugyan) [inline src/routes/member/[studentId]/edit-bio/+page.server.ts:29]: formData validation seems a bit repetitive. Do you think we should use zod for form validation?

File: src/routes/member/[studentId]/profile-picture/+page.server.ts [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+        .extract({
+          left: Number(x ?? "0"),
+          top: Number(y ?? "0"),
+          width: Number(width),
+          height: Number(height),
+        })
+        // resize to MAX 400x400
+        .resize(400, 400, {
+          fit: "cover",
+          withoutEnlargement: true,
```
Comment (danieladugyan) [inline src/routes/member/[studentId]/profile-picture/+page.server.ts:116]: Maybe image should be enlarged?

File: src/lib/components/ClassBadge.svelte [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    BME: "border-[#fafaf9] text-[#fafaf9]",
+    Dokt: "border-[#c0c0c0] text-[#c0c0c0]",
+    "?": "border-gray-400 text-gray-400",
+  };
+</script>
+
+{#if member.classProgramme && member.classYear}
+  <span
+    class="badge badge-outline
+      {size === 'sm' ? 'badge-sm text-xs font-light' : 'badge-lg text-xl font-semibold'}
```
[the remark this answers is not in the document]
Comment (Macludde) [reply src/lib/components/ClassBadge.svelte:20]: no

File: src/lib/components/socials/MemberAvatar.svelte [hunk 6]
```diff
@@ -2,17 +2,18 @@
...
   export let member: Pick<Member, "picturePath" | "classProgramme" | "classYear">;
-  export let size: "sm" | "md" | "lg" | "xl" = "md";
+  export let size: "sm" | "md" | "lg" | "xl" | null = "md";
   const sizeToWidth = {
     sm: "w-4",
     md: "w-8",
     lg: "w-12",
-    xl: "w-16",
+    xl: "w-32",
   };
```
[the remark this answers is not in the document]
Comment (Macludde) [reply src/lib/components/socials/MemberAvatar.svelte:11]: How?

File: src/routes/member/[studentId]/+page.server.ts (hunk 1 above)
Comment (Macludde) [reply src/routes/member/[studentId]/+page.server.ts:10]: okidoki

File: src/routes/member/[studentId]/edit-bio/+page.server.ts (hunk 3 above)
Comment (Macludde) [reply src/routes/member/[studentId]/edit-bio/+page.server.ts:29]: Or [Superform](https://superforms.rocks/)

File: src/routes/member/[studentId]/profile-picture/+page.server.ts (hunk 4 above)
Comment (Macludde) [reply src/routes/member/[studentId]/profile-picture/+page.server.ts:116]: no I don't think so. It will just take more space on the server without any more quality. The frontend will scale it anyway

File: src/routes/member/[studentId]/edit-bio/+page.server.ts (hunk 3 above)
Comment (Macludde) [reply src/routes/member/[studentId]/edit-bio/+page.server.ts:29]: I think this can be changed in a future pr

File: src/routes/NavBar.svelte [hunk 7]
```diff
@@ -37,7 +37,7 @@
       {#if $page.data.session}
         <button class="btn btn-neutral" on:click={() => signOut()}> Logga ut </button>
         {#if user?.student_id}
-          <a class="btn btn-ghost" href="/member/{user.student_id}">
+          <a class="btn btn-ghost" href="/members/{user.student_id}">
```
Comment (danieladugyan) [inline src/routes/NavBar.svelte:40]: Did we revert the change to `member`?

Comment (Macludde) [reply src/routes/NavBar.svelte:40]: Yes, because the old site uses /members/ and it's hard-coded into for example comments, and I could guess for example news articles

**Review by danieladugyan — APPROVED** [review]
